pyproject.toml

```diff
@@ -0,0 +1,12 @@
+[build-system]
+requires = [
+    "setuptools >= 42",
+    "setuptools_scm[toml] >= 6.2",
+    "setuptools_scm_git_archive",
+    "wheel >= 0.29.0",
+]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools_scm]
+write_to = "src/fparser/_version.py"
+git_describe_command = "git describe --dirty --tags --long --first-parent"
```

Comment:
Also, the setup.py had:
```
MAJOR = 0
MINOR = 0
MICRO = 16
```
why does it show a 0.0.014 version now? Is this right?

Reply:
If fact if I don `pip install -e .` I get an error messge with yet another version:
`psyclone 2.3.1 requires fparser==0.0.16, but you have fparser 0.0.15.dev19+gadc4126 which is incompatible.`

But everything works after it, so maybe is not an issue?

Reply:
I suspect you need to fetch the latest git tags: `git fetch --tags`

Reply:
Just on how the version number works: `setuptools_scm` takes the output of the `git describe` command and parses it.

The scheme is [explained in their README](https://github.com/pypa/setuptools_scm#default-versioning-scheme), but essentially if it's exactly on a tag, then the version number is just the tag (so on the next release, it will be `0.0.17`). Otherwise, it will bump the patch version and add the `.devN+g<commit>` as a prerelease version.

Reply:
Thanks @ZedThree. I too get various version numbers from that `git describe` command. If I drop `--first-parent` then it does give me a "0.0.16"-based string. From looking at their README (https://github.com/pypa/setuptools_scm#configuration-parameters) I see that setting `git_describe_command` is described as: "Use with caution, this is a function for advanced use". What is the reason you set it originally?

Reply:
I originally set this as there were some issues with the default command, but it looks like these have been fixed upstream, so you can safely delete this line.

With the current HEAD of this branch, after deleting that line, you should get `fparser-0.0.17.dev215+g2fc7918.d20230313`

Reply:
Great, thanks @ZedThree . I've removed that line now.